### PR TITLE
chore: upgrade angular 12 to 13

### PR DESCRIPTION
**Pull request by cseas**
## Description

[Jira][6]

Angular 12 has reached end of life.

Steps followed:
- Enabled legacy-peer-deps because upgrade was throwing dependency issues. This is a problem because latest node/npm is stricter than node 14 & below. Suggestion was to downgrade node or enable this flag.
- [Upgrade angular][1] core & cli packages
- Upgrade angular material
- All lint objects were deleted from `angular.json` (which were being used for TSLint config)

Build broken, tests broken, lint broken

- Fixed typescript errors manually, build fixed
- Now updated test related packages manually
- Removed transformers from jest config, not required anymore for [jest-preset-angular][3]
- Jest 27 had a lot breaking changes because it changes test runner from `jest-jasmine2` to `jest-circus`
- Most tests started breaking because they were written Jasmine code
- Ran jest codemod for migrating jasmine globals to jest
- Manually fixed tests that were still breaking
- The only tests that are modified are `explorer.component.test.ts`
- One package upgrade was missed: `@angular/flex-layout`
- After upgrading this, bunch of more typescript errors, manually fixed

Build worked, tests worked, lint broken

- Run `ng lint` which told me to run `@angular-eslint/schematics`
- Removed [solution style tsconfig][4] which was preventing execution of eslint schematics package
- This installed all eslint related packages
- Then added eslint using [CLI][2] to each child project
- That initialised eslint for all children projects and updated `lint` keys in `angular.json`
- Manually fixed lint errors

Build fixed, tests fixed, eslint fixed, prettier failing the lint check

- Changed prettier script in `package.json` to check only projects and src folders
- Added script to fix prettier issues automatically
- Fixed file formatting with `npm run prettier:fix`

Build, test, lint, all passing. Ready to merge.

Stacked PRs: 
- Update angular versions in peerDependencies of children projects: 
  #123 
- Remove all TSLint related code, configs and packages:
  #124

**Future work:** Upgrade to angular 14, upgrade hyperdash & disable peer deps from npmrc

### Testing
- [x] All tests passing
- [x] All lints passing
- [x] Builds without errors
- [x] Tested locally
- [x] Tested on [stage][5]

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

[1]: https://update.angular.io/?v=12.0-13.0
[2]: https://github.com/angular-eslint/angular-eslint/issues/632#issuecomment-1035180154
[3]: https://github.com/thymikee/jest-preset-angular/blob/main/CHANGELOG.md#breaking-changes-5
[4]: https://github.com/hypertrace/hypertrace-ui/issues/1976
[5]: https://hypertrace.concierge.stage.razorpay.in/home?time=1h
[6]: https://razorpay.atlassian.net/browse/OBSV-1213

### CODE REVIEW

**Comment by cseas** [comment]
Blocked on this issue: https://github.com/angular-eslint/angular-eslint/issues/790#issuecomment-1366648324
Update: Resolved

**Comment by cseas** [comment]
Build and tests fixed. Kinda blocked on this to fix lint: https://github.com/angular/angular-cli/issues/24477
Update: Resolved using a manual workaround

**Comment by cseas** [comment]
Hyperdash has a dependency on Angular 12. 
A PR is open for upgrading to Angular 14: https://github.com/hypertrace/hyperdash-angular/pull/649
Need to keep legacy-peer-deps flag enabled in npmrc till that upgrade is done.

File: .eslintrc.json
Comment (jaywalker21) [inline .eslintrc.json]: Should be `"prefix": "ht"` here, so even if someone adds a new directive in the future, this rule is validated correctly.

File: projects/assets-library/.eslintrc.json
Comment (jaywalker21) [inline projects/assets-library/.eslintrc.json]: This should be `ht`

File: projects/assets-library/.eslintrc.json
Comment (jaywalker21) [inline projects/assets-library/.eslintrc.json]: This should be `ht`

File: projects/dashboards/.eslintrc.json
Comment (jaywalker21) [inline projects/dashboards/.eslintrc.json]: Let's change this to `ht`

File: projects/graphql-client/.eslintrc.json
Comment (jaywalker21) [inline projects/graphql-client/.eslintrc.json]: Let's change both prefixes in this file to `ht`

File: projects/test-utils/.eslintrc.json
Comment (jaywalker21) [inline projects/test-utils/.eslintrc.json]: Let's change prefixes to `ht`

**Review by jaywalker21 — COMMENTED** [review]
Minor comments, rest seems all right

File: projects/test-utils/.eslintrc.json
Comment (cseas) [reply projects/test-utils/.eslintrc.json]: Changed all prefix selectors rules to "ht" :tada:

**Review by jaywalker21 — APPROVED** [review]
LGTM, please hold on to merging to master till the `micro` second changes are done with.

**Comment by cseas** [comment]
Hypertrace has been put in maintenance mode by the org. Closing this for now. 

Not deleting the branch as this upgrade might be needed if security team raises an issue about Hypertrace running on EoL Angular version.

**Pull request closed without merging**
